src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -857,11 +857,15 @@ public void ResetTimeout()
 
         private async Task TimerLoop(TimerAwaitable timer)
         {
+            // tell the server you intend to ping
+            await SendHubMessage(PingMessage.Instance);
+            //await SendCoreAsyncCore(PingMessage.Instance);
+
             // initialize the timers
-            timer.Start();
             ResetSendPing();
             ResetTimeout();
-            
+            timer.Start();
```

Comment:
Is there a reason you moved Start afterwards? Add a one sentence comment saying why it should happen after the reset methods.

Reply:
The order of these lines is purely visual / a matter of taste. Mostly wanted the two ping-related lines to be adjacent because it seemed nicer to read.